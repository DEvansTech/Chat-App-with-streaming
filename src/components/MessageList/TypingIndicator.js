import React from 'react';
import PropTypes from 'prop-types';

import styled from '@stream-io/styled-components';

import { StreamChat } from 'stream-chat';

import { Avatar } from '../Avatar';
import { themed } from '../../styles/theme';
import { withTranslationContext } from '../../context';

const TypingText = styled.Text`
  margin-left: 10px;
  font-size: ${({ theme }) => theme.typingIndicator.text.fontSize}px;
  color: ${({ theme }) => theme.typingIndicator.text.color};
  ${({ theme }) => theme.typingIndicator.text.css};
`;

const Container = styled.View`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: flex-start;
  ${({ theme }) => theme.typingIndicator.container.css};
`;

class TypingIndicator extends React.PureComponent {
  static themePath = 'typingIndicator';
  static propTypes = {
    typing: PropTypes.object,
    client: PropTypes.instanceOf(StreamChat),
    /**
     * Defaults to and accepts same props as: [Avatar](https://getstream.github.io/stream-chat-react-native/#avatar)
     */
    Avatar: PropTypes.oneOfType([PropTypes.node, PropTypes.elementType]),
  };
  static defaultProps = {
    Avatar,
  };

  constructTypingString = (dict) => {
    const { client, t } = this.props;
    const arr2 = Object.keys(dict);
    const arr3 = [];
    arr2.forEach((item, i) => {
      if (client.user.id === dict[arr2[i]].user.id) {
        return;
      }
      arr3.push(dict[arr2[i]].user.name || dict[arr2[i]].user.id);
    });

    let outStr = '';
    if (arr3.length === 1) {
      outStr = t('{{ user }} is typing...', { user: arr3[0] });
      dict;
    } else if (arr3.length === 2) {
      //joins all with "and" but =no commas
      //example: "bob and sam"
      outStr = t('{{ firstUser }} and {{ secondUser }} are typing...', {
        firstUser: arr3[0],
        secondUser: arr3[1],
      });
    } else if (arr3.length > 2) {
      //joins all with commas, but last one gets ", and" (oxford comma!)
      //example: "bob, joe, and sam"
      outStr = t('{{ commaSeparatedUsers }} and {{ lastUser }} are typing...', {
        commaSeparatedUsers: arr3.slice(0, -1).join(', '),
        lastUser: arr3.slice(-1),
      });
    }

    return outStr;
  };

  render() {
    const { typing, Avatar, client } = this.props;
    const typingUsers = Object.values(typing);

    return (
      <Container>
        {typingUsers
          .filter(({ user }) => user.id !== client.user.id)
          .map(({ user }, idx) => (
            <Avatar
              image={user.image}
              size={24}
              name={user.name || user.id}
              key={user.id + idx}
            />
          ))}
        <TypingText>{this.constructTypingString(typing)}</TypingText>
      </Container>
    );
  }
}

export default withTranslationContext(themed(TypingIndicator));
