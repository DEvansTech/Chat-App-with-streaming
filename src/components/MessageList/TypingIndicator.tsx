import React from 'react';

import { useTypingString } from './hooks/useTypingString';

import DefaultAvatar, { AvatarProps } from '../Avatar/Avatar';

import { useChannelContext } from '../../contexts/channelContext/ChannelContext';
import { useChatContext } from '../../contexts/chatContext/ChatContext';
import { styled } from '../../styles/styledComponents';
import { themed } from '../../styles/theme';

import type {
  DefaultAttachmentType,
  DefaultChannelType,
  DefaultCommandType,
  DefaultEventType,
  DefaultMessageType,
  DefaultReactionType,
  DefaultUserType,
} from '../../types/types';

const Container = styled.View`
  align-items: center;
  flex-direction: row;
  justify-content: flex-start;
  ${({ theme }) => theme.typingIndicator.container.css};
`;

const TypingText = styled.Text`
  color: ${({ theme }) => theme.typingIndicator.text.color};
  font-size: ${({ theme }) => theme.typingIndicator.text.fontSize}px;
  margin-left: 10px;
  ${({ theme }) => theme.typingIndicator.text.css};
`;

export type TypingIndicatorProps = {
  /**
   * Defaults to and accepts same props as: [Avatar](https://getstream.github.io/stream-chat-react-native/#avatar)
   */
  Avatar?: React.ComponentType<Partial<AvatarProps>>;
};

const TypingIndicator = <
  At extends Record<string, unknown> = DefaultAttachmentType,
  Ch extends Record<string, unknown> = DefaultChannelType,
  Co extends string = DefaultCommandType,
  Ev extends Record<string, unknown> = DefaultEventType,
  Me extends Record<string, unknown> = DefaultMessageType,
  Re extends Record<string, unknown> = DefaultReactionType,
  Us extends DefaultUserType = DefaultUserType
>({
  Avatar = DefaultAvatar,
}: TypingIndicatorProps) => {
  const { typing } = useChannelContext<At, Ch, Co, Ev, Me, Re, Us>();
  const { client } = useChatContext<At, Ch, Co, Ev, Me, Re, Us>();
  const typingString = useTypingString<At, Ch, Co, Ev, Me, Re, Us>();

  const typingUsers = Object.values(typing);

  return (
    <Container testID='typing-indicator'>
      {typingUsers
        .filter(({ user }) => user?.id !== client?.user?.id)
        .map(({ user }, idx) => (
          <Avatar
            image={user?.image}
            key={`${user?.id}${idx}`}
            name={user?.name || user?.id}
            size={24}
            testID={`typing-avatar-${idx}`}
          />
        ))}
      <TypingText>{typingString}</TypingText>
    </Container>
  );
};

TypingIndicator.themePath = 'typingIndicator';

export default themed(TypingIndicator) as typeof TypingIndicator;
