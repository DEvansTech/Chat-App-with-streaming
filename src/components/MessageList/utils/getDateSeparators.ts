import type { PaginatedMessageListContextValue } from '../../../contexts/paginatedMessageListContext/PaginatedMessageListContext';
import type { ThreadContextValue } from '../../../contexts/threadContext/ThreadContext';
import type {
  DefaultAttachmentType,
  DefaultChannelType,
  DefaultCommandType,
  DefaultEventType,
  DefaultMessageType,
  DefaultReactionType,
  DefaultUserType,
  UnknownType,
} from '../../../types/types';

export type GetDateSeparatorsParams<
  At extends UnknownType = DefaultAttachmentType,
  Ch extends UnknownType = DefaultChannelType,
  Co extends string = DefaultCommandType,
  Ev extends UnknownType = DefaultEventType,
  Me extends UnknownType = DefaultMessageType,
  Re extends UnknownType = DefaultReactionType,
  Us extends UnknownType = DefaultUserType
> = {
  messages:
    | PaginatedMessageListContextValue<At, Ch, Co, Ev, Me, Re, Us>['messages']
    | ThreadContextValue<At, Ch, Co, Ev, Me, Re, Us>['threadMessages'];
  hideDateSeparators?: boolean;
};

export type DateSeparators = {
  [key: string]: Date;
};

export const getDateSeparators = <
  At extends UnknownType = DefaultAttachmentType,
  Ch extends UnknownType = DefaultChannelType,
  Co extends string = DefaultCommandType,
  Ev extends UnknownType = DefaultEventType,
  Me extends UnknownType = DefaultMessageType,
  Re extends UnknownType = DefaultReactionType,
  Us extends UnknownType = DefaultUserType
>(
  params: GetDateSeparatorsParams<At, Ch, Co, Ev, Me, Re, Us>,
) => {
  const { hideDateSeparators, messages } = params;
  const dateSeparators: DateSeparators = {};

  if (hideDateSeparators) {
    return dateSeparators;
  }

  for (let i = 0; i < messages.length; i++) {
    const previousMessage = messages[i - 1];
    const message = messages[i];

    if (
      message.type === 'channel.event' ||
      message.type === 'message.read' ||
      message.deleted_at
    ) {
      continue;
    }

    const messageDate = message.created_at.getDay();

    const prevMessageDate = previousMessage
      ? previousMessage.created_at.getDay()
      : messageDate;

    if (i === 0 || messageDate !== prevMessageDate) {
      dateSeparators[message.id] = message.created_at;
    }
  }

  return dateSeparators;
};
