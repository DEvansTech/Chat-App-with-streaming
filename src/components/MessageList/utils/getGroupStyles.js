export const getGroupStyles = (messagesWithDates, noGroupByUser) => {
  const numberOfMessages = messagesWithDates.length;
  const messageGroupStyles = {};

  const messages = [...messagesWithDates];

  for (let i = 0; i < numberOfMessages; i++) {
    const previousMessage = messages[i - 1];
    const message = messages[i];
    const nextMessage = messages[i + 1];
    const groupStyles = [];

    if (message.type === 'channel.event') {
      continue;
    }

    if (message.type === 'message.date') {
      continue;
    }

    const userId = message.user ? message.user.id : null;

    const isTopMessage =
      !previousMessage ||
      previousMessage.type === 'message.date' ||
      previousMessage.type === 'system' ||
      previousMessage.type === 'channel.event' ||
      (previousMessage.attachments &&
        previousMessage.attachments.length !== 0) ||
      userId !== previousMessage.user.id ||
      previousMessage.type === 'error' ||
      previousMessage.deleted_at;

    const isBottomMessage =
      !nextMessage ||
      nextMessage.type === 'message.date' ||
      nextMessage.type === 'system' ||
      nextMessage.type === 'channel.event' ||
      (nextMessage.attachments && nextMessage.attachments.length !== 0) ||
      userId !== nextMessage.user.id ||
      nextMessage.type === 'error' ||
      nextMessage.deleted_at;

    if (isTopMessage) {
      groupStyles.push('top');
    }

    if (isBottomMessage) {
      if (isTopMessage || message.deleted_at || message.type === 'error') {
        groupStyles.splice(0, groupStyles.length);
        groupStyles.push('single');
      } else {
        groupStyles.push('bottom');
      }
    }

    if (!isTopMessage && !isBottomMessage) {
      if (message.deleted_at || message.type === 'error') {
        groupStyles.splice(0, groupStyles.length);
        groupStyles.push('single');
      } else {
        groupStyles.splice(0, groupStyles.length);
        groupStyles.push('middle');
      }
    }

    if (message.attachments.length !== 0) {
      groupStyles.splice(0, groupStyles.length);
      groupStyles.push('single');
    }

    if (noGroupByUser) {
      groupStyles.splice(0, groupStyles.length);
      groupStyles.push('single');
    }

    messageGroupStyles[message.id] = groupStyles;
  }

  return messageGroupStyles;
};
