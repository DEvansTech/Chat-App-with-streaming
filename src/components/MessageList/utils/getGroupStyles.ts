import type { DateSeparators } from './getDateSeparators';

import type { GroupType } from '../hooks/useMessageList';

import type { PaginatedMessageListContextValue } from '../../../contexts/paginatedMessageListContext/PaginatedMessageListContext';
import type { ThreadContextValue } from '../../../contexts/threadContext/ThreadContext';
import type {
  DefaultAttachmentType,
  DefaultChannelType,
  DefaultCommandType,
  DefaultEventType,
  DefaultMessageType,
  DefaultReactionType,
  DefaultUserType,
  UnknownType,
} from '../../../types/types';
import type { FormatMessageResponse } from 'stream-chat';

export type GetGroupStylesParams<
  At extends UnknownType = DefaultAttachmentType,
  Ch extends UnknownType = DefaultChannelType,
  Co extends string = DefaultCommandType,
  Ev extends UnknownType = DefaultEventType,
  Me extends UnknownType = DefaultMessageType,
  Re extends UnknownType = DefaultReactionType,
  Us extends UnknownType = DefaultUserType
> = {
  dateSeparators: DateSeparators;
  messages:
    | PaginatedMessageListContextValue<At, Ch, Co, Ev, Me, Re, Us>['messages']
    | ThreadContextValue<At, Ch, Co, Ev, Me, Re, Us>['threadMessages'];
  hideDateSeparators?: boolean;
  maxTimeBetweenGroupedMessages?: number;
  noGroupByUser?: boolean;
  userId?: string;
};

export const getGroupStyles = <
  At extends UnknownType = DefaultAttachmentType,
  Ch extends UnknownType = DefaultChannelType,
  Co extends string = DefaultCommandType,
  Ev extends UnknownType = DefaultEventType,
  Me extends UnknownType = DefaultMessageType,
  Re extends UnknownType = DefaultReactionType,
  Us extends UnknownType = DefaultUserType
>(
  params: GetGroupStylesParams<At, Ch, Co, Ev, Me, Re, Us>,
) => {
  const {
    dateSeparators,
    hideDateSeparators,
    maxTimeBetweenGroupedMessages,
    messages,
    noGroupByUser,
    userId,
  } = params;
  const messageGroupStyles: { [key: string]: GroupType[] } = {};

  const messagesFilteredForNonUser = messages.filter(
    (message) => !message.deleted_at || userId === message.user?.id,
  );

  for (let i = 0; i < messagesFilteredForNonUser.length; i++) {
    const previousMessage = messagesFilteredForNonUser[i - 1] as
      | typeof messagesFilteredForNonUser[0]
      | undefined;
    const message = messagesFilteredForNonUser[i];
    const nextMessage = messagesFilteredForNonUser[i + 1] as
      | typeof messagesFilteredForNonUser[0]
      | undefined;
    const groupStyles: GroupType[] = [];

    const userId = message?.user?.id || null;

    const isTopMessage =
      !previousMessage ||
      previousMessage.type === 'system' ||
      userId !== previousMessage?.user?.id ||
      previousMessage.type === 'error' ||
      !!previousMessage.deleted_at ||
      (!hideDateSeparators && dateSeparators[message.id]) ||
      messageGroupStyles[previousMessage.id]?.includes('bottom');

    const isBottomMessage =
      !nextMessage ||
      nextMessage.type === 'system' ||
      userId !== nextMessage?.user?.id ||
      nextMessage.type === 'error' ||
      !!nextMessage.deleted_at ||
      (!hideDateSeparators && dateSeparators[nextMessage.id]) ||
      (maxTimeBetweenGroupedMessages !== undefined &&
        nextMessage.created_at.getTime() - message.created_at.getTime() >
          maxTimeBetweenGroupedMessages);

    /**
     * Add group styles key for top message
     */
    if (isTopMessage) {
      groupStyles.push('top');
    }

    /**
     * Add group styles key for bottom message
     */
    if (isBottomMessage) {
      /**
       * If the bottom message is also the top, or deleted, or an error,
       * add the key for single message instead of bottom
       */
      if (isTopMessage || message.deleted_at || message.type === 'error') {
        groupStyles.splice(0, groupStyles.length);
        groupStyles.push('single');
      } else {
        groupStyles.push('bottom');
      }
    }

    /**
     * Add the key for all non top or bottom messages, if the message is
     * deleted or an error add the key for single otherwise middle
     */
    if (!isTopMessage && !isBottomMessage) {
      if (message.deleted_at || message.type === 'error') {
        groupStyles.splice(0, groupStyles.length);
        groupStyles.push('single');
      } else {
        groupStyles.splice(0, groupStyles.length);
        groupStyles.push('middle');
      }
    }

    /**
     * If noGroupByUser set add the key for single
     */
    if (noGroupByUser) {
      groupStyles.splice(0, groupStyles.length);
      groupStyles.push('single');
    }

    if (message.id) {
      messageGroupStyles[message.id] = groupStyles;
    }
  }

  return messageGroupStyles;
};
