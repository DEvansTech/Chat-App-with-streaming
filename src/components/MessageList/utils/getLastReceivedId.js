export const getLastReceivedId = (messages) => {
  let lastReceivedId = null;

  for (let i = messages.length; i > 0; i--) {
    if (
      messages[i] !== undefined &&
      messages[i].status !== undefined &&
      messages[i].status === 'received'
    ) {
      lastReceivedId = messages[i].id;
      break;
    }
  }

  return lastReceivedId;
};
