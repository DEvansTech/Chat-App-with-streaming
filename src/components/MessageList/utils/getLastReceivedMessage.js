export const getLastReceivedMessage = (messages) => {
  let lastReceivedMessage = null;

  /**
   * There are no status on dates so they will be skipped
   */
  for (let i = 0; i < messages.length; i++) {
    if (
      messages[i] !== undefined &&
      messages[i].id !== undefined &&
      messages[i].status !== undefined &&
      messages[i].status === 'received'
    ) {
      lastReceivedMessage = messages[i];
      break;
    }
  }

  return lastReceivedMessage;
};
