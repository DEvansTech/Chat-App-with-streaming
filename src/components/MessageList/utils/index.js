export { getGroupStyles } from './getGroupStyles';
export { getLastReceivedMessage } from './getLastReceivedMessage';
export { getReadStates } from './getReadStates';
export { insertDates } from './insertDates';
