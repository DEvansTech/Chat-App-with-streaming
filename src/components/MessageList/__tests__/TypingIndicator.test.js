import React from 'react';
import { View } from 'react-native';
import { cleanup, render, waitFor } from '@testing-library/react-native';
import {
  generateStaticUser,
  generateUser,
  getTestClientWithUser,
} from 'mock-builders';

import { Chat } from '../../Chat';
import { ChannelContext, TranslationContext } from '../../../context';
import TypingIndicator from '../TypingIndicator';
import { Streami18n } from '../../../utils';

afterEach(cleanup);

describe('TypingIndicator', () => {
  let chatClient;

  it('should render typing indicator for two users', async () => {
    const t = jest.fn((key) => key);
    const user0 = generateUser();
    const user1 = generateUser();
    const user2 = generateUser();

    chatClient = await getTestClientWithUser({ id: user0.id });
    await chatClient.setUser(user0, 'testToken');
    const typing = { user1: { user: user1 }, user2: { user: user2 } };

    const { getByTestId } = render(
      <Chat client={chatClient}>
        <TranslationContext.Provider value={{ t }}>
          <ChannelContext.Provider value={{ typing }}>
            <TypingIndicator />
          </ChannelContext.Provider>
        </TranslationContext.Provider>
      </Chat>,
    );
    expect(t).toHaveBeenCalledWith(
      '{{ firstUser }} and {{ secondUser }} are typing...',
      {
        firstUser: user1.name,
        secondUser: user2.name,
      },
    );
    await waitFor(() => {
      expect(getByTestId('typing-indicator')).toBeTruthy();
    });
  });

  it('should render typing indicator for one user', async () => {
    const t = jest.fn((key) => key);
    const user0 = generateUser();
    const user1 = generateUser();

    chatClient = await getTestClientWithUser({ id: user0.id });
    await chatClient.setUser(user0, 'testToken');
    const typing = { user1: { user: user1 } };

    const { getByTestId } = render(
      <Chat client={chatClient}>
        <TranslationContext.Provider value={{ t }}>
          <ChannelContext.Provider value={{ typing }}>
            <TypingIndicator />
          </ChannelContext.Provider>
        </TranslationContext.Provider>
      </Chat>,
    );
    expect(t).toHaveBeenCalledWith('{{ user }} is typing...', {
      user: user1.name,
    });
    await waitFor(() => {
      expect(getByTestId('typing-indicator')).toBeTruthy();
    });
  });

  it('should render typing Avatar for one user', async () => {
    const t = jest.fn((key) => key);
    const user0 = generateUser();
    const user1 = generateUser();

    chatClient = await getTestClientWithUser({ id: user0.id });
    await chatClient.setUser(user0, 'testToken');
    const typing = { user1: { user: user1 } };

    const { getByTestId, queryByTestId } = render(
      <Chat client={chatClient}>
        <TranslationContext.Provider value={{ t }}>
          <ChannelContext.Provider value={{ typing }}>
            <TypingIndicator Avatar={View} />
          </ChannelContext.Provider>
        </TranslationContext.Provider>
      </Chat>,
    );
    expect(t).toHaveBeenCalledWith('{{ user }} is typing...', {
      user: user1.name,
    });

    await waitFor(() => {
      expect(getByTestId('typing-indicator')).toBeTruthy();
      expect(getByTestId('typing-avatar-0')).toBeTruthy();
      expect(queryByTestId('typing-avatar-1')).toBeFalsy();
    });
  });

  it('should render typing Avatar for two users', async () => {
    const t = jest.fn((key) => key);
    const user0 = generateUser();
    const user1 = generateUser();
    const user2 = generateUser();

    chatClient = await getTestClientWithUser({ id: user0.id });
    await chatClient.setUser(user0, 'testToken');
    const typing = { user1: { user: user1 }, user2: { user: user2 } };

    const { getByTestId } = render(
      <Chat client={chatClient}>
        <TranslationContext.Provider value={{ t }}>
          <ChannelContext.Provider value={{ typing }}>
            <TypingIndicator Avatar={View} />
          </ChannelContext.Provider>
        </TranslationContext.Provider>
      </Chat>,
    );

    await waitFor(() => {
      expect(getByTestId('typing-indicator')).toBeTruthy();
      expect(getByTestId('typing-avatar-0')).toBeTruthy();
      expect(getByTestId('typing-avatar-1')).toBeTruthy();
    });
  });

  it('should render no typing Avatars', async () => {
    const t = jest.fn((key) => key);
    const user0 = generateUser();

    chatClient = await getTestClientWithUser({ id: user0.id });
    await chatClient.setUser(user0, 'testToken');
    const typing = {};

    const { getByTestId, queryByTestId } = render(
      <Chat client={chatClient}>
        <TranslationContext.Provider value={{ t }}>
          <ChannelContext.Provider value={{ typing }}>
            <TypingIndicator Avatar={View} />
          </ChannelContext.Provider>
        </TranslationContext.Provider>
      </Chat>,
    );

    await waitFor(() => {
      expect(getByTestId('typing-indicator')).toBeTruthy();
      expect(queryByTestId('typing-avatar-0')).toBeFalsy();
      expect(queryByTestId('typing-avatar-1')).toBeFalsy();
    });
  });

  it('should match typing indicator snapshot', async () => {
    const i18nInstance = new Streami18n();
    const { t } = await i18nInstance.getTranslators();
    const user0 = generateStaticUser(0);
    const user1 = generateStaticUser(1);
    const user2 = generateStaticUser(3);

    chatClient = await getTestClientWithUser({ id: user0.id });
    await chatClient.setUser(user0, 'testToken');
    const typing = { user1: { user: user1 }, user2: { user: user2 } };

    const { toJSON } = render(
      <Chat client={chatClient}>
        <TranslationContext.Provider value={{ t }}>
          <ChannelContext.Provider value={{ typing }}>
            <TypingIndicator />
          </ChannelContext.Provider>
        </TranslationContext.Provider>
      </Chat>,
    );
    await waitFor(() => {
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
