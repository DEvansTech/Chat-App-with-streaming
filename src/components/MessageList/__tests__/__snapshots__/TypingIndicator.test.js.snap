// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypingIndicator should match typing indicator snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      },
    ]
  }
  testID="typing-indicator"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
      ]
    }
  >
    <Image
      accessibilityLabel="initials"
      onError={[Function]}
      resizeMethod="resize"
      size={24}
      source={
        Object {
          "uri": "https://i.imgur.com/SLx06PP.png",
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 12,
            "height": 24,
            "width": 24,
          },
        ]
      }
      testID="avatar-image"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
      ]
    }
  >
    <Image
      accessibilityLabel="initials"
      onError={[Function]}
      resizeMethod="resize"
      size={24}
      source={
        Object {
          "uri": "https://i.imgur.com/T68W8nR_d.webp?maxwidth=728&fidelity=grand",
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 12,
            "height": 24,
            "width": 24,
          },
        ]
      }
      testID="avatar-image"
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "rgba(0,0,0,0.5)",
          "fontSize": 14,
          "marginLeft": 10,
        },
      ]
    }
  >
    Grant and Neil are typing...
  </Text>
</View>
`;
